### PR TITLE
Update dependencies to fix security vulnerabilities

### PR DESCRIPTION
**Pull request by pradeepnschrodinger**
## Description

The total list of security vulnerabilities can be found [here](https://github.com/schrodinger/fixed-data-table-2/security/dependabot?q=is%3Aopen+sort%3Aseverity) (ordered by severity).

Dependabot has already pulled up a few [active PRs](https://github.com/schrodinger/fixed-data-table-2/pulls/app%2Fdependabot) to fix some of them. 
The ones that don't have an automatic PR made yet needs to be resolved manually because it either requires
* a major bump in dependencies which _might_ introduce breaking changes or extra code changes
* or there's an existing and conflicting dependency with the vulnerable package, making it difficult to directly bump it without breaking other packages 

The changes in this PR tackles _most_ of the remaining packages/vulnerabilities. 
Once these are merged in, Dependabot should take another look at the repo, and update the security vulnerabilities list.

## Motivation and Context

* Ticket in JIRA: [SS-35747](https://jira.schrodinger.com/browse/SS-35747)
* Security vulnerabilities reported by Dependabot: https://github.com/schrodinger/fixed-data-table-2/pulls/app%2Fdependabot

## How Has This Been Tested?

Tested through the local webpack server, and also tested across the examples, in both production and dev mode.
**TODO: Run it through LD** 

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.

### CODE REVIEW

**Comment by novakps** [comment]
lgtm.  @mycrobe do we have/need a 'business-as-usual' dependency update ticket for FDT2?

File: build_helpers/startSiteDevServer.sh [hunk 1]
```diff
@@ -5,6 +5,7 @@ PATH=$(npm bin):$PATH
 rm -rf ./__site__
 rm -rf ./__site_prerender__
 ./build_helpers/buildAPIDocs.sh
+webpack --config "$PWD/site/webpack-client.config.js"
 webpack --config "$PWD/site/webpack-prerender.config.js"
 ./build_helpers/buildSiteIndexPages.sh
-webpack-dev-server --config "$PWD/site/webpack-client.config.js" --no-info --content-base __site__ --port ${1-8080}
+webpack-dev-server --config "$PWD/site/webpack-client.config.js"
```
Comment (pradeepnschrodinger) [inline build_helpers/startSiteDevServer.sh:11]: content base is no longer a parameter to the CLI, but should now be customized inside the webpack config through the `devServer.static` property

File: build_helpers/startSiteDevServer.sh [hunk 2]
```diff
@@ -5,6 +5,7 @@ PATH=$(npm bin):$PATH
 rm -rf ./__site__
 rm -rf ./__site_prerender__
 ./build_helpers/buildAPIDocs.sh
+webpack --config "$PWD/site/webpack-client.config.js"
```
Comment (pradeepnschrodinger) [inline build_helpers/startSiteDevServer.sh:8]: Slightly unrelated to this PR, but I found out that some of the output files necessary for the `buildSiteIndexPages.sh` script below wasn't being generated in production mode. So we run webpack before calling the script to generate it.

File: site/webpack-client.config.js [hunk 3]
```diff
@@ -1,7 +1,7 @@
 var path = require('path');
 var webpack = require('webpack');
 var MiniCssExtractPlugin = require('mini-css-extract-plugin');
-var UglifyJsPlugin = require('uglifyjs-webpack-plugin');
+const TerserPlugin = require('terser-webpack-plugin');
```
Comment (pradeepnschrodinger) [inline site/webpack-client.config.js:4]: `uglifyjs-webpack-plugin` is deprecated by `terser-webpack-plugin` (see notes: https://github.com/webpack-contrib/uglifyjs-webpack-plugin#deprecated)

File: src/plugins/Scrollbar.js [hunk 4]
```diff
@@ -28,7 +28,7 @@ var UNSCROLLABLE_STATE = {
   scrollable: false,
 };
 
-var FACE_MARGIN = parseInt(cssVar('scrollbar-face-margin'), 10);
+var FACE_MARGIN = parseInt(cssVar('--scrollbar-face-margin'), 10);
```
Comment (pradeepnschrodinger) [inline src/plugins/Scrollbar.js:31]: The latest version of `postcss-custom-properties` requires you to specify variables with `--` prefixed

File: src/stubs/cssVar.js [hunk 5]
```diff
@@ -14,16 +14,16 @@
...
-  'scrollbar-face-color': '#c2c2c2',
-  'scrollbar-face-margin': '4px',
-  'scrollbar-face-radius': '6px',
-  'scrollbar-size': '15px',
-  'scrollbar-size-large': '17px',
-  'scrollbar-track-color': '#fff',
-  'border-color': '#d3d3d3',
-  'fbui-white': '#fff',
-  'fbui-desktop-background-light': '#f6f7f8',
+  '--scrollbar-face-active-color': '#7d7d7d',
```
Comment (pradeepnschrodinger) [inline src/stubs/cssVar.js:17]: The latest version of `postcss-custom-properties` requires you to specify variables with `--` prefixed

**Comment by pradeepnschrodinger** [comment]
Released with [v1.2.3](https://github.com/schrodinger/fixed-data-table-2/releases/tag/v1.2.3)

**Pull request merged**